pkg/nsx/services/nsxserviceaccount/cluster.go

```diff
@@ -249,8 +249,14 @@ func (s *NSXServiceAccountService) createPIAndCCP(normalizedClusterName string,
 			return "", err
 		}
 		s.PrincipalIdentityStore.Add(&pi)
-	} else if !hasPI != (piObj == nil) {
+	} else if !hasPI && (piObj == nil) {
```

Comment:
No need to change this, this branch is to handle same NSXSA name with different UID.

Reply:
done